mazda/installer/config/androidauto/jci/gui/apps/_androidauto/templates/AndroidAuto/js/AndroidAutoTmplt.js

```diff
@@ -23,6 +23,17 @@ function AndroidAutoTmplt(uiaId, parentDiv, templateID, controlProperties)
 AndroidAutoTmplt.prototype.handleControllerEvent = function(eventID)
 {
     log.debug('handleController() called, eventID: ' + eventID);
+	switch(eventID) {
+		
+		case "ccw":
+		case "up":
+			document.getElementById("aaStatusText").scrollTop -= 100;

```

Comment:
a note on defensive programming -- you might want to make sure scrollTop doesn't exceed some min/max values. what happens if I keep scrolling up or down?

Reply:
Nope the DOM property 'scrollTop' will reset itself to the max value if it is exceeded and also back to 0 if it becomes a negative number.

Reply:
Checked to make sure https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTop

Reply:
oh, rock on then!

Reply:
hey maybe you can look at this other thing.  I am trying to figure out how to pass the keystrokes over EVIOCGRAB to control audio when AA doesn't have audio focus something like this:
```cpp
bool hasAudioFocus = callbacks->audioFocus;
 ...
 ...
case KEY_LEFTBRACE:
    printf("KEY_LEFTBRACE\n");
    if (hasAudioFocus) {
      scanCode = HUIB_NEXT;
    } else {
      scanCode = KEY_LEFTBRACE;
    }
    break;
case KEY_RIGHTBRACE:
    printf("KEY_RIGHTBRACE\n");
    if (hasAudioFocus) {
      scanCode = HUIB_PREV;
    } else {
      scanCode = KEY_RIGHTBRACE;
    }
    break;
```
its obvious why that densest work but I can't really find a way to do it. maybe `close(kbd_fd)` for like a quick second but that looks like it wont work right.  I'm looking for some way to just pass the keystroke along to the gui.

Reply:
that is in mazda/outputs.cpp BTW